extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -61,6 +64,10 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    this.pageAttachmentEl_ = this.element
+      .closest('amp-story-page')
+      .querySelector('amp-story-page-attachment');
```

Comment:
I think we should put this within an if block that only executes if `this.pageAttachmentEl_` is null, to ensure that `.closest()` and `.querySelector()` only need to be called once

Reply:
Does it not run once since it's in `layoutCallback`?

Reply:
Can tags be added without a shopping attachment? (in terms of being valid AMP, or being an intended scenario)

Reply:
That's not intended UX. Good catch. 
Wrote #37580 for tracking that.